examples/templates/devcontainer-kubernetes/main.tf

```diff
@@ -260,49 +266,19 @@ resource "kubernetes_deployment" "main" {
           image             = var.cache_repo == "" ? local.devcontainer_builder_image : envbuilder_cached_image.cached.0.image
           image_pull_policy = "Always"
           security_context {}
-          env {
-            name  = "CODER_AGENT_TOKEN"
-            value = coder_agent.main.token
-          }
-          env {
-            name  = "CODER_AGENT_URL"
-            value = data.coder_workspace.me.access_url
-          }
-          env {
-            name  = "ENVBUILDER_GIT_URL"
-            value = local.repo_url
-          }
-          env {
-            name  = "ENVBUILDER_INIT_SCRIPT"
-            value = coder_agent.main.init_script
-          }
-          env {
-            name  = "ENVBUILDER_FALLBACK_IMAGE"
-            value = data.coder_parameter.fallback_image.value
-          }
-          env {
-            name  = "ENVBUILDER_CACHE_REPO"
-            value = var.cache_repo
-          }
-          env {
-            name  = "ENVBUILDER_PUSH_IMAGE"
-            value = var.cache_repo == "" ? "" : "true"
-          }
-          env {
-            name  = "ENVBUILDER_INSECURE"
-            value = var.insecure_cache_repo
-          }
-          env {
-            name  = "ENVBUILDER_DOCKER_CONFIG_BASE64"
-            value = try(data.kubernetes_secret.cache_repo_dockerconfig_secret[0].data[".dockerconfigjson"], "")
+
+          # Set the environment using cached_image.cached.0.env if the cache repo is enabled.
+          # Otherwise, use the local.envbuilder_env.
+          # You could alternatively write the environment variables to a ConfigMap or Secret
+          # and use that as `env_from`.
+          dynamic "env" {
+            for_each = nonsensitive(var.cache_repo == "" ? local.envbuilder_env : envbuilder_cached_image.cached.0.env_map)
```

Comment:
review: apparently this is required to be able to `for_each` over a `map` :thinking:

Reply:
Yes. I previously used it here. 
https://github.com/coder/modules/blob/4c45d69994937700f3033f120a489e1c17c1afd6/hcp-vault-secrets/main.tf#L69

Reply:
This makes sense since `resource "envbuilder_cached_image" "cached"` has count set to 1.